xwiki-pro-macros-api/src/main/java/com/xwiki/macros/showhideif/internal/macro/AbstractShowHideIfMacro.java

```diff
@@ -125,6 +126,30 @@ protected boolean doesMatch(ShowHideIfMacroParameters parameters)
             default:
                 break;
         }
+
+        if (parameters.getDisplayType() != null
+            && parameters.getDisplayType() != ShowHideIfMacroParameters.DisplayType.NONE)
+        {
+            boolean isExportPrintable = "export".equalsIgnoreCase(xcontext.getAction());
+            switch (parameters.getDisplayType()) {
+                case DEFAULT:
+                    if (isExportPrintable) {
+                        matchAllRes = false;
+                    } else {
+                        matchAnyRes = true;
+                    }
+                    break;
+                case PRINTABLE:
+                    if (isExportPrintable) {
+                        matchAnyRes = true;
+                    } else {
+                        matchAllRes = false;
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
```

Comment:
the values of `matchAllRes` and `matchAnyRes` seem to be ovewritten. Is that intended?  This suggests that the values of the `authenticationType` and `groupReferenceList` params are ignored. Or that the they are taken in consideration in the following order 1.DisplayType 2.AuthType 3.GroupRefList. If so, this should be better specified in the parameter description so it will be clear for the users.